src/test/java/com/example/spring/service/BookServiceTest.java

```diff
@@ -0,0 +1,123 @@
+package com.example.spring.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.example.spring.dto.BookDto;
+import com.example.spring.mapper.BookMapper;
+import com.example.spring.models.Book;
+import com.example.spring.repositories.book.BookRepository;
+import com.example.spring.services.BookServiceImpl;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Get all books from store")
+    void getAll_GivenBooksInStore_ShouldReturnAllBooks() {
+        // Given
+        Book book1 = new Book();
+        book1.setId(1L);
+        book1.setTitle("Book 1");
+        book1.setAuthor("Author 1");
+        book1.setIsbn("1111111111");
+        book1.setPrice(BigDecimal.valueOf(10.99));
+        book1.setDescription("Description 1");
+        book1.setCoverImage("cover_image_1.jpg");
```

Comment:
I suggest to to create some class for example TestObjectMother or TestUtil and put there logic of creation of mock data to make test classes more readable